spec/treasury_and_analytics.md

```diff
@@ -0,0 +1,135 @@
+| quantity | BigInt | Number of tokens: MUST be greater than or equal to 0: MUST support BigInt |
+| source | String | The url where this row's value was obtained, e.g., https://coinmarketcap.com/currencies/bitdao/ | 
+
+#### Spot price specification:
+
+| Field | Type | Description |
+| ---- | ---- | ---- |
+| yyyy | number | Year of retrieved value: MUST equal 2021 |
+| mm | number | Month of retrieved value: MUST be between 1 and 12 |
+| dd | number | Day of retrieved value: MUST be between 1 and 31 (validation tbd) |
+| token | String | Short token identifier, e.g., BIT, BTC, ETH, USDC, USDT |
+| price | Decimal | Price of token: must accommodate exactly 2 decimal places, greater than 0, less than 5,000 |
+| source | String | The url where this row's value was obtained, e.g., https://coinmarketcap.com/currencies/bitdao/ | 
+
+## Building daily static page
+
+Procedure to build static page:
+
+1. Retrieve token spot price and store on S3
+   1. Copy content of existing data file and store with a new name
+2. Retrieve token amounts from BitDAO treasury
+    1. Copy content of existing data file and store with a new name
+3. Generate the static html page with links pointing to the latest data files
+4. Include analytics javascript library to allow online aggregation 
+
+## Treasury, Holdings
+
+Display current ERC-20 token balance of the following coins:
+
+- BIT
+- USDT
+- USDC
+- xSUSHI
+
+## Treasury, Inbound
+
+- Should support reconciliation
+- Reconciliation should not change past events and their computation
+    - historical values should be hard coded
+- Should support changing expected contribution amounts, e.g,. 50/50 BTC to ETH vs more tokens.
+    - BitDAO has [pledged](https://docs.bitdao.io/partners/bybit-pledge) 50% ETH, 25% USDT, 25% USDC
+- Day start = UTC 0, Day end = UTC 0
+- USDT and USDC price will be $1.00 for simplicity
+- Other Asset price feed based on Index Price (currently the weighted average of Kraken, Gemini,
+  Coinbase Pro, Bittrex, and Bitstamp spot pairs) as described
+  here https://help.bybit.com/hc/en-us/articles/360039261094
+- Expected value sources can vary, currently includes:
+    - ByBit
+        - Futures products only (linear and inverse perpetual contracts, quarterly futures
+          contracts, etc.)
+        - Initial Contributed Asset mix: 50% ETH, 25% USDT, 25% USDC
+        - Daily trade volume as provided by Bybit API to CoinGecko
+        - Daily trade volume converted to units of each Contributed Asset at end of day's CoinGecko
+          price
+        - Contribution start date: July 15, 2021
+        - At the start, the injections will be monthly. Based on the cumulative units of each
+          Contributed Asset (derived from the daily calculations).
+        - Contributions from Bybit will come from the following wallets:
+          0x6a159503e49ddB8E49659D6CACf5236aC4213D86
+
+### An Example
+
+- Day trade volume $10,000,000,000
```

Comment:
question: Determining trade volume from the contract could be difficult, unless those details are available from an Oracle. Might be easier with input from the pledger (either directly stating volume, or using a Reality.eth) 🤔

Reply:
The trade volume is from coingecko, see the following:
> Daily trade volume as provided by Bybit API to CoinGecko

Reply:
Ah, retrieving the data from an off-chain source, makes sense 👍